tensorflow/lite/micro/test_helpers.cc

```diff
@@ -142,13 +156,34 @@ ModelBuilder::Node ModelBuilder::AddNode(
   return next_operator_id_ - 1;
 }
 
+void ModelBuilder::AddMetadata(const char* description_string,
+                            const int32_t* metadata_buffer_data,
+                            size_t num_elements) {
+  metadata_[ModelBuilder::nbr_of_metadata_buffers_] =
+              CreateMetadata(*builder_,
+                             builder_->CreateString(description_string),
+                             1 + ModelBuilder::nbr_of_metadata_buffers_);
+
+  metadata_buffers_[nbr_of_metadata_buffers_] = tflite::CreateBuffer(*builder_,
+                                                                       builder_->CreateVector((uint8_t*)metadata_buffer_data,
+                                                                       sizeof(uint32_t) * num_elements));
+
+  ModelBuilder::nbr_of_metadata_buffers_++;
+}
+
 const Model* ModelBuilder::BuildModel(
     std::initializer_list<ModelBuilder::Tensor> inputs,
     std::initializer_list<ModelBuilder::Tensor> outputs) {
   // Model schema requires an empty buffer at idx 0.
-  constexpr size_t kBufferSize = 1;
-  const flatbuffers::Offset<Buffer> buffers[kBufferSize] = {
-      tflite::CreateBuffer(*builder_)};
+  size_t kBufferSize = 1 + ModelBuilder::nbr_of_metadata_buffers_;
+  flatbuffers::Offset<Buffer> buffers[kBufferSize];
```

Comment:
This is dynamic sized array. Not sure if it's allowed on all platforms. 

Alternatively you might also consider other approaches:
- create a large enough list of size kMaxMetadataBuffers?
- append the first two elements to metadata_buffers_ in the constructor.